### PR TITLE
Move include/text.php to class structure

### PR DESCRIPTION
**Pull request by nupplaphil**
Part of #3878 

I moved all `include/text.php` functions into proper methods.
Each commit is one function, I moved.
For almost every moved function, I created some tests.

Functions, which I removed completely because of no usage:
- `return_bytes()`
- `attribute_contains()`
- `qp()`

### CODE REVIEW

**Comment by annando** [comment]
I must confess, I still don't get the advantage of using non-static class functions that need to have that baseobject thingy instead of using static class functions. (Especially when using functions that don't need initialization)

For me it simply makes coding more complicated.

**Comment by nupplaphil** [comment]
It becomes better once we can use it directly with the constructor. Than we don't need the whole `self::getClass()` construct anymore.. It'sna compromise between both worlds..

**Comment by annando** [comment]
I don't think that this "constructor" thing would be easier for me. The reality is that there are fewer and fewer places in the code where I'm finding myself able of contributing something. For example the logger class is still a mystery for me.

**Comment by nupplaphil** [comment]
Don't forget https://github.com/friendica/friendica-addons/pull/904

**Comment by MrPetovan** [comment]
I'm with @annando, I don't think we should use instantiated objects for helper methods without dependencies. I believe it is a waste of lines, memory and expectations.

**Comment by nupplaphil** [comment]
The problem about static classes - even for utils - is their global state.
A real example: the `DateTimeFormat`.

For our current cache implementation, I'd like to test the TTL. In the code, we use `DateTimeFormat::utc()`. But at any cost, and this is good for production, it returns the current timestamp. 

Currently there are three possibiltis:
- I've to set down the TTL to a small value, so the test isn't blocked. But this can lead to other side effects because of too small TTL (which is indeed the case..) and I don't test a "real" config. And the last constraint about it is that the value is an integer, so mininum two seconds sleep for each run (6 caches, 3 assertions = slow down 26 seconds)
- I can mock the `DateTimeFormat::utc()` globally, but I've to do it for the whole phpunit call (= every test case). So it's impossible to test the `utc()` method itself anymore and I've to take the whole execution path of the API tests into account..
- I've to start an own process for each cache test and mock Datetimeformat for these processes. This is possible too but forces PHPUnit to create own process containers for each test. This leads to long execution duration and more memory consumption. And the phpunit doc says one should do it just if there's no other chance..

If we use an instance even for utils, we can override these class at every call. Either just use it at is or mock the whole class or just parts of the class.Or we speedup the method because we return predefined fixtures in some situations. So the `DateTimeFormat::utc()` method returns the same timestamp for each call which makes testing more deterministic too.

File: src/Protocol/Activity.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Friendica\Protocol;
+
+/**
+ * Base class for the Activity namespace
+ */
+final class Activity
```
[the remark this answers is not in the document]
Comment (MrPetovan) [reply src/Protocol/Activity.php]: Ha ha, forgot about that.

[the remark this answers is not in the document]
Comment (nupplaphil) [reply src/Protocol/Activity.php]: I'll create a Follow-up, I'm in the middle of this change :)))

[the remark this answers is not in the document]
Comment (nupplaphil) [reply src/Protocol/Activity.php]: Done by #7770

**Pull request merged**